cmake/modules/GetVersionFromGit.cmake

```diff
@@ -0,0 +1,187 @@
+# the following cases:
+#
+# - If the current Git checkout points to an exact Git tag,
+# we assume it is an officially released version (either a release or a pre-release).
+#
+# - If we are in a release branch (with the suffix `release/`),
+# we assume that each commit (i.e., each development build) represents a potential new patch version.
+#
+# - If we are in any other branch,
+# we assume that each commit (i.e., each development build) represents a potential new minor version.
+#
+# - In any other case (for example, if there's no git information available),
+# version `0.0.0` will be returned as a fallback.
+#
+# For the officially released versions, we assume the Git tag will carry on the correct information,
+# so we return it as-is.
+#
+# For development versions, the string is built as follows:
+# `<x>.<y>.<z>-<count>+<commit>[suffix]`
+# Where:
+# - `<x>.<y>.<z>` represents the next version number, reflecting either a patch for release branches
+# or a minor version for development branches.
+# - `<count>` is the number of commits ahead from either:
+# - the latest tag on the branch, for release branches; or
+# - the closest common ancestor with the branch holding the latest tagged version,
+# for development branches.
+# - `<commit>` refers to the first 7 digits of the commit hash.
+# - `[suffix]` the value of `match_suffix`, if any.
+# Note: all non-alphanumerics will be converted to hyphens.
+function(get_version_from_git _var match_suffix exclude_suffix)
+    set(_options "")
+
+    if(match_suffix)
+        set(_options "--match=*${match_suffix}")
+    endif()
+
+    if(exclude_suffix)
+        set(_options "${_options} --exclude=*${exclude_suffix}")
+    endif()
+
+    # Release version
+    # Try to obtain the exact git tag
+    _falcosecurity_execute_git(tag describe HEAD --exact-match ${_options})
+
+    if(tag)
+        # A tag has been found: use it as the libs version
+        set(${_var}
+            "${tag}"
+            PARENT_SCOPE)
+        return()
+    endif()
+
+    # Otherwise, we are on a dev version
+    _falcosecurity_execute_git(current_hash rev-parse HEAD)
+    _falcosecurity_execute_git(current_hash_short rev-parse --short=7 HEAD)
+    _falcosecurity_execute_git(current_branch rev-parse --abbrev-ref HEAD)
+
+    set(dev_version "0.0.0")
+
+    # dev version / in a `release/M.m.x` branch
+    if(current_branch MATCHES "${FALCOSECURITY_RELEASE_BRANCH_REGEX}")
+        # get the latest tag on the release branch
+        set(_options --match "*.*.*${match_suffix}")
+
+        if(exclude_suffix)
+            set(_options ${_options} --exclude "*${exclude_suffix}")
+        endif()
+
+        _falcosecurity_execute_git(tag describe --tags ${_options} --abbrev=0 "${current_branch}")
+
+        if(tag)
+            _falcosecurity_extract_version("${tag}" x y z match_found)
+
+            if(match_found)
+                math(EXPR z_plus_one "${z} + 1")
+                set(dev_version "${x}.${y}.${z_plus_one}")
```

Comment:
uhm this means that on the same release branch, we can tag only patch versions for drivers tags `+driver`... it seems reasonable in the end we shouldn't introduce new changes in the schema in a release branch

Reply:
Correct. This should also be documented in RELASE.md.

Reply:
I've also updated RELEASE.md.